flake.nix

```diff
@@ -1,19 +1,53 @@
 {
   description = "Deku development environment";
-  inputs.nixpkgs.url = "github:nixos/nixpkgs";
+  inputs.nixpkgs.url = "github:nixos/nixpkgs/nixos-unstable";
   inputs.flake-utils.url = "github:numtide/flake-utils";
-  inputs.esy-fhs.url = "github:d4hines/esy-fhs";
-  inputs.esy-fhs.inputs.nixpkgs.follows = "nixpkgs";
-  outputs = { self, nixpkgs, flake-utils, esy-fhs }:
+  inputs.ocaml-overlays.url = "github:anmonteiro/nix-overlays";
+  inputs.ocaml-overlays.inputs.nixpkgs.follows = "nixpkgs";
+
+  outputs = { self, nixpkgs, flake-utils, ocaml-overlays }:
     flake-utils.lib.eachDefaultSystem (system:
-      let pkgs = import nixpkgs { inherit system; }; in
+      let pkgs = import nixpkgs { inherit system; overlays = [ocaml-overlays.overlay]; };
+      in
       {
-        defaultApp = esy-fhs.lib.makeFHSApp {
-          inherit system;
-          extraPackages = with pkgs; [
-            ligo
+        devShell = (pkgs.mkShell {
+          shellHook = ''
+            npm install
+            export PATH="node_modules/.bin:_build/default/src/bin:$PATH"
+          '';
+
+          packages = with pkgs.ocaml-ng.ocamlPackages_multicore; [
+            ocaml
+            dune_2
+            findlib
+            ocaml-lsp
+            ocamlformat-rpc
+            pkgs.nodejs
+            pkgs.nodePackages.npm
           ];
-        };
+
+          propagatedBuildInputs = with pkgs.ocaml-ng.ocamlPackages_multicore; [
```

Comment:
I think that is awesome! If I understand correctly, everyone using `nix` will use the exact same version of package, and these versions will also be the same on our CI?

Reply:
Yes, that's the goal, fully reproducible development environments :)